packages/preview-middleware-client/src/adp/controllers/ExtensionPoint.controller.ts

```diff
@@ -85,25 +85,46 @@ export default class ExtensionPoint extends BaseDialog {
         }
 
         this.model.setProperty('/extensionPointName', extensionPointName);
+        const hasDefaultContent = this.hasDefaultContentForName(extensionPointName);
+        this.model.setProperty('/hasDefaultContent', hasDefaultContent);
+    }
+
+    /**
+     * Determines whether there is default content associated with the given name.
+     *
+     * @param {string} name - The name to check for associated default content.
+     * @returns {boolean} - True if there is non-empty default content associated with the name, false otherwise.
+     */
+    private hasDefaultContentForName(name: string): boolean {
+        const defaultContent = this.data?.info.find((v) => v.name === name)?.defaultContent;
+        return (defaultContent && defaultContent.length > 0) || false;
     }
```

Comment:
We can return directly `defaultContent?.length > 0`

Reply:
In this case, defaultContent can be undefined so I still need to check whether we have it and then check its length. But I changed it a little here: [c86f608](https://github.com/SAP/open-ux-tools/pull/1867/commits/c86f608325dd2d9f9b520beffc14d1a6122242c5)